src/main/java/graphql/ExecutionInput.java

```diff
@@ -245,27 +268,84 @@ public Builder localContext(Object localContext) {
         }
 
         /**
-         * By default you will get a {@link GraphQLContext} object but you can set your own.
+         * The legacy context object
          *
          * @param context the context object to use
+         *
          * @return this builder
+         *
+         * @deprecated - the {@link ExecutionInput#getGraphQLContext()} is a fixed mutable instance now
          */
+        @Deprecated
         public Builder context(Object context) {
             this.context = context;
             return this;
         }
 
+        /**
+         * The legacy context object
+         *
+         * @param contextBuilder the context builder object to use
+         *
+         * @return this builder
+         *
+         * @deprecated - the {@link ExecutionInput#getGraphQLContext()} is a fixed mutable instance now
+         */
+        @Deprecated
         public Builder context(GraphQLContext.Builder contextBuilder) {
             this.context = contextBuilder.build();
             return this;
         }
 
+        /**
+         * The legacy context object
+         *
+         * @param contextBuilderFunction the context builder function to use
+         *
+         * @return this builder
+         *
+         * @deprecated - the {@link ExecutionInput#getGraphQLContext()} is a fixed mutable instance now
+         */
+        @Deprecated
         public Builder context(UnaryOperator<GraphQLContext.Builder> contextBuilderFunction) {
             GraphQLContext.Builder builder = GraphQLContext.newContext();
             builder = contextBuilderFunction.apply(builder);
             return context(builder.build());
         }
 
+        /**
+         * This will give you a builder of {@link GraphQLContext} and any values you set will be copied
+         * into the underlying {@link GraphQLContext} of this execution input
+         *
+         * @param builderFunction a builder function you can use to put values into the context
+         *
+         * @return this builder
+         */
+        public Builder graphQLContext(Consumer<GraphQLContext.Builder> builderFunction) {
+            GraphQLContext.Builder builder = GraphQLContext.newContext();
+            builderFunction.accept(builder);
+            this.graphQLContext.putAll(builder);
+            return this;
+        }
+
+        /**
+         * This will put all the valuyes from the map into the underlying {@link GraphQLContext} of this execution input
```

Comment:
```suggestion
         * This will put all the values from the map into the underlying {@link GraphQLContext} of this execution input
```

Reply:
yeah I think this is doing what you want - its a new context here but that is "copied / putAll" into the one context object